### PR TITLE
Support vschema_scan_node.

### PR DESCRIPTION
**Pull request by taberylyang**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/vschema_scan_node.cpp [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+        VLOG_ROW << "VSchemaScanNode output rows: " << block->rows();
+    } while (block->rows() == 0 && !(*eos));
+    return Status::OK();
+}
+
+Status VSchemaScanNode::write_slot_to_vectorized_column(void* slot, 
+                                                        SlotDescriptor* slot_desc,
+                                                        vectorized::MutableColumnPtr* column_ptr) {
+    vectorized::MutableColumnPtr* col_ptr = column_ptr;
+    if (slot_desc->is_nullable()) {
```
Comment (HappenLee) [inline be/src/vec/exec/vschema_scan_node.cpp:124]: schema scan会出现null的逻辑吗

File: be/src/vec/exec/vschema_scan_node.cpp [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+            // tuple project
+            project_tuple();
+
+            for (int i = 0; i < _slot_num; ++i) {
+                auto slot_desc = _dest_tuple_desc->slots()[i];
+                if (!slot_desc->is_materialized()) {
+                    continue;
+                }
+
+                if (_dest_single_tuple->is_null(slot_desc->null_indicator_offset())) {
```
Comment (HappenLee) [inline be/src/vec/exec/vschema_scan_node.cpp:85]: 这里为什么直接用_src_single_tuple来copy数据，而是通过_dest_single_tuple倒腾了一手？

File: be/src/vec/exec/vschema_scan_node.cpp [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+
+        case TYPE_DATE: {
+            DateTimeValue* ts_slot = reinterpret_cast<DateTimeValue*>(slot);
+            reinterpret_cast<vectorized::ColumnVector<vectorized::Int128>*>(col_ptr)->insert_data(
+                reinterpret_cast<char*>(&ts_slot), 0);
+            break;
+        }
+
+        case TYPE_DATETIME: {
+            DateTimeValue* ts_slot = reinterpret_cast<DateTimeValue*>(slot);
```
Comment (HappenLee) [inline be/src/vec/exec/vschema_scan_node.cpp:196]: 这里使用一下新的VecDateTimeValue吧

File: be/src/vec/exec/vschema_scan_node.cpp [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+        for (int i = 0; i < _slot_num; ++i) {
+            if (mem_reuse) {
+                columns[i] = std::move(*block->get_by_position(i).column).mutate();
+            } else {
+                columns[i] = _dest_tuple_desc->slots()[i]->get_empty_mutable_column();
+            }
+        }
+        while (true) {
+            RETURN_IF_CANCELLED(state);
+            int batch_size = state->batch_size();
```
Comment (HappenLee) [inline be/src/vec/exec/vschema_scan_node.cpp:63]: 只用一次就没必要新加一个变量了，直接columns[0]->size() == state->batch_size（）就行了

File: be/src/vec/exec/vschema_scan_node.cpp (hunk 1 above)
Comment (taberylyang) [reply be/src/vec/exec/vschema_scan_node.cpp:124]: 目前看确实会出现null值。

File: be/src/vec/exec/vschema_scan_node.cpp (hunk 2 above)
Comment (taberylyang) [reply be/src/vec/exec/vschema_scan_node.cpp:85]: _src_single_tuple返回的是系统表的全部列，_dest_single_tuple返回的是用户查询和过滤所需的列。通过project_tuple函数，对_src_single_tuple进行投影得到了_dest_single_tuple。这里要填充最后用户的查询结果，所以使用的是_dest_single_tuple。

**Pull request merged**